pkg/cmd/roachtest/registry/test_spec.go

```diff
@@ -135,6 +135,9 @@ type TestSpec struct {
 	// If one is not specified, the default behavior is to upload
 	// a binary with the crdb_test flag randomly enabled or disabled.
 	CockroachBinary ClusterCockroachBinary
+
+	// RequiresDeprecatedWorkload, if true, will validate that `workload` exists.
+	RequiresDeprecatedWorkload bool
```

Comment:
Now that you've added this to the test spec., it would be nice to remove the redundant `c.PutE(ctx, t.L(), t.DeprecatedWorkload(), "./workload")` of the opted-in test, and replace it with the single (upload) step in the testRunner, akin to what `c.PutCockroach(ctx, l, t)` does, for every test.